cmd/bucket-replication.go

```diff
@@ -1572,25 +1577,23 @@ func NewReplicationPool(ctx context.Context, o ObjectLayer, opts replicationPool
 		workers = WorkerAutoDefault
 		failedWorkers = MRFWorkerAutoDefault
 	}
+
 	pool := &ReplicationPool{
-		replicaCh:               make(chan ReplicateObjectInfo, 100000),
-		replicaDeleteCh:         make(chan DeletedObjectReplicationInfo, 100000),
-		mrfReplicaCh:            make(chan ReplicateObjectInfo, 100000),
-		workerKillCh:            make(chan struct{}, workers),
-		mrfWorkerKillCh:         make(chan struct{}, failedWorkers),
-		existingReplicaCh:       make(chan ReplicateObjectInfo, 100000),
-		existingReplicaDeleteCh: make(chan DeletedObjectReplicationInfo, 100000),
-		resyncer:                newresyncer(),
-		mrfSaveCh:               make(chan MRFReplicateEntry, 100000),
-		saveStateCh:             make(chan struct{}, 1),
-		ctx:                     ctx,
-		objLayer:                o,
-		priority:                priority,
-	}
-
-	pool.ResizeWorkers(workers)
+		workers:         make([]chan ReplicationWorkerOperation, 0, workers),
+		existingWorkers: make(chan ReplicationWorkerOperation, 100000),
+		mrfReplicaCh:    make(chan ReplicationWorkerOperation, 100000),
+		mrfWorkerKillCh: make(chan struct{}, failedWorkers),
+		resyncer:        newresyncer(),
+		mrfSaveCh:       make(chan MRFReplicateEntry, 100000),
+		saveStateCh:     make(chan struct{}, 1),
+		ctx:             ctx,
+		objLayer:        o,
+		priority:        priority,
+	}
+
+	pool.ResizeWorkers(workers, 0)
 	pool.ResizeFailedWorkers(failedWorkers)
-	go pool.AddExistingObjectReplicateWorker()
+	go pool.AddWorker(pool.existingWorkers, nil)
```

Comment:
needs pool.workerWg.Add(1) here to avoid panic

```
panic: sync: negative WaitGroup counter

goroutine 869 [running]:
sync.(*WaitGroup).Add(0x2486900?, 0xc0005580c0?)
	/usr/local/go/src/sync/waitgroup.go:83 +0xda
sync.(*WaitGroup).Done(0x0?)
	/usr/local/go/src/sync/waitgroup.go:108 +0x25
github.com/minio/minio/cmd.(*ReplicationPool).AddWorker(0xc003d66780, 0xc002089320, 0xc003d66780)
	/home/kp/code/src/github.com/minio/minio/cmd/bucket-replication.go:1638 +0x3d6
created by github.com/minio/minio/cmd.(*ReplicationPool).ResizeWorkers
	/home/kp/code/src/github.com/minio/minio/cmd/bucket-replication.go:1693 +0xe5
```